stdlib/public/core/UnicodeScalarProperties.swift

```diff
@@ -49,23 +49,96 @@ extension Unicode.Scalar {
   }
 }
 
-/// Boolean properties that are defined by the Unicode Standard (i.e., not
-/// ICU-specific).
 extension Unicode.Scalar.Properties {
-  internal func _hasBinaryProperty(
-    _ property: __swift_stdlib_UProperty
-  ) -> Bool {
-    return __swift_stdlib_u_hasBinaryProperty(icuValue, property) != 0
+  // This OptionSet represents the 64 bit integer value returned when asking
+  // '_swift_stdlib_getBinaryProperties' where each bit indicates a unique
+  // Unicode defined binary property of a scalar. For example, bit 8 represents
+  // the 'isAlphabetic' property for scalars.
+  //
+  // Note: If one ever updates this list, be it reordering bits adding new ones,
+  // etc., please update the same list found in:
+  // 'utils/gen-unicode-data/Sources/GenScalarProps/BinProps.swift'.
+  internal struct _BinaryProperties: OptionSet {
+    let rawValue: UInt64
+
+    init(_ rawValue: UInt64) {
+      self.rawValue = rawValue
+    }
+
+    // Because we defined the labelless init, we lose the memberwise one
+    // generated, so define that here to satisfy the 'OptionSet' requirement.
+    init(rawValue: UInt64) {
+      self.rawValue = rawValue
+    }
+
+    static var changesWhenCaseFolded       : Self { Self(1 &<< 0) }
+    static var changesWhenCaseMapped       : Self { Self(1 &<< 1) }
+    static var changesWhenLowercased       : Self { Self(1 &<< 2) }
+    static var changesWhenNFKCCaseFolded   : Self { Self(1 &<< 3) }
+    static var changesWhenTitlecased       : Self { Self(1 &<< 4) }
+    static var changesWhenUppercased       : Self { Self(1 &<< 5) }
+    static var isASCIIHexDigit             : Self { Self(1 &<< 6) }
+    static var isAlphabetic                : Self { Self(1 &<< 7) }
+    static var isBidiControl               : Self { Self(1 &<< 8) }
+    static var isBidiMirrored              : Self { Self(1 &<< 9) }
```

Comment:
I actually don't really mind these, either! 😌 I agree that the tabular form is nicer here; it helps spot inconsistencies long lists such as this one. (But regular formatting would look fine to me, too.)

(What I would (usually) object to is either applying such formatting to a list that only has a small handful of items, or introducing a formal rule to _require_ these custom alignments. This is because I imagine it would be difficult to reliably implement them in a fully automated code formatting/normalization tool, which I'm still hoping we'll introduce some day. 🤞)